pallets/remarks/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+	dispatch::GetDispatchInfo,
+	pallet_prelude::*,
+	traits::{IsSubType, OriginTrait},
+	BoundedVec,
+};
+use frame_system::pallet_prelude::*;
+pub use pallet::*;
+use sp_runtime::traits::Dispatchable;
+use sp_std::boxed::Box;
+pub use weights::WeightInfo;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+mod weights;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The overarching call type.
+		type RuntimeCall: Parameter
+			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
+			+ GetDispatchInfo
+			+ From<frame_system::Call<Self>>
+			+ IsSubType<Call<Self>>
+			+ IsType<<Self as frame_system::Config>::RuntimeCall>;
+
+		/// Weight information.
+		type WeightInfo: WeightInfo;
+
+		/// The type attached to the remark event.
+		type Remark: Parameter + Member + Default;
+
+		/// Type that restrains the maximum remarks that can be attached to a
+		/// call.
+		type MaxRemarks: Get<u32>;
```

Comment:
Nit: Maybe we want the name to be more explicit by adding _per call_
```suggestion
		type MaxRemarksPerCall: Get<u32>;
```

Reply:
Done, thank you!